File: webapp-bff/src/backend/chat/dto/group-chat-response.dto.ts
import { IsString, ValidateNested } from "class-validator";
import { Type } from "class-transformer";
import { ChatMessageResponseDto } from "./chat-message-response.dto";
import { GroupChatMemberResponseDto } from "./group-chat-member-resonse.dto";

export class GroupChatResponseDto {
  @IsString()
  groupId: string;

  @ValidateNested({ each: true })   // list of nested DTOs
  @Type(() => ChatMessageResponseDto)
  messages: ChatMessageResponseDto[];

  @ValidateNested({ each: true })   // list of nested DTOs
  @Type(() => GroupChatMemberResponseDto)
  members: GroupChatMemberResponseDto[];
}

File: chat-service/src/main/java/com/tompang/carpool/chat_service/message/dto/GroupChatResponseDto.java
package com.tompang.carpool.chat_service.message.dto;

import java.util.List;

import lombok.Builder;
import lombok.Data;

@Data
@Builder
public class GroupChatResponseDto {
    private String groupId;
    private List<ChatMessageResponseDto> messages;
    private List<GroupMemberDto> members;
}
